File: projects/m1/016-next-day/solutions/enricozoboli/solution_next_day.py
year = int(input("Insert the year: "))
month = int(input("Insert the month: "))
day = int(input("Insert the day: "))

next_year = year
next_month = month
next_day = day

if month == 2:
    if(
        (year % 400 == 0 or (year % 100 != 0 and year % 4 == 0))
        and day == 28
    ):
        next_day += 1

    elif day == 29 or day == 28:
        next_day = 1
        next_month += 1
    else:
        next_day += 1

elif (
    month == 1 or month == 3 or month == 5 or 7 <= month <= 8
    or month == 10
):
    if day == 31:
        next_day = 1
        next_month += 1
    else :
        next_day += 1

elif month == 12:
    if day == 31:
        next_day = 1
        next_month = 1
        next_year += 1
    else:
        next_day += 1

elif month == 4 or month == 6 or month == 9 or month == 11:
    if day == 30:
        next_day = 1
        next_month +=1
    else:
        next_day += 1
    

print(f"{next_year}-{next_month}-{next_day}")

File: projects/m1/016-next-day/solutions/valentinaBazzacchi/solution-next-day.js
let year = prompt("Insert a year in number!");
let month = prompt("Insert a month in number!"); 
let day = prompt("Insert a day in number!"); 

// condizione mesi da 30 giorni => true or false 
let days30Month;
if (month == 4 || month == 6 || month == 9 || month == 11) {
days30Month = true;
} else {
days30Month = false;
}

// condizione mesi da 31 giorni => true or false 
let days31Month;
if (!days30Month && month != 2) {
days31Month = true;
} else {
days31Month = false; 
}

// condizione scatto anno => true or false
let nextYear
if (month == 12 && day == 31) {
nextYear = true; 
} else {
nextYear = false;  
}

// condizione anno bisestile => true or false
let leapYear;
if ((year % 400 == 0) || year % 100 != 0 && year % 4 == 0) {
leapYear = true;
} else {
leapYear = false;    
}
 
// condizione scatto mese => true or false
let nextMonth;
if (month == 2 && ((day == 28 && !leapYear) || (day == 29 && leapYear))) {
nextMonth = true;
} else if ((day == 31 && days31Month) || (day == 30 && days30Month)) {
nextMonth = true;
} else {
nextMonth = false;         
} 
 
// riporto la data successiva combinando le diverse condizioni
if (!leapYear && month == 2 && day > 28) {
alert(year + " isn't a leap year!");
} else if (days30Month && day > 30){
alert("This month only has 30 days!"); 
} else if (!nextYear && !nextMonth) {
alert(year + " - " + ("00" + month).slice(-2) + " - " + ("00" + (+day + 1)).slice(-2));
} else if (!nextYear && nextMonth) {
alert(year + " - " + ("00" + (+month + 1)).slice(-2) + " - 01");
} else if (nextYear) {
alert((+year + 1) + " - 01 - 01");
} 